File: PSET1/cash.c
#include <stdio.h>
#include <math.h>
#include <cs50.h>

float dollars;
int coins = 0;

int main(void)
{
    //Prompt user for Cash owned
    do
    {
        dollars = get_float("Cash owed: ");
    }
    while (dollars <= 0);

    //Rounding the cents to the nearest penny
    int cents = round(dollars * 100);

    //Deducting the values from the bigger to the smaller
    while (cents > 0)
    {
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else
        {
            cents --;
        }
        coins++;
    }
    //Print the minimal amount of coins
    printf("%i\n", coins);
}

File: PSET6/cash.py
# prototype
def main():
    owed = get_owed()
    coins = calc_coins(owed)
    print(f"{coins}")


# prompting the user for the cash owned
def get_owed():
    while True:
        try:
            owed = float(input("Cash owed: "))
            if owed > 0:
                break
        except ValueError:
            None

    return owed


# calculating the change
def calc_coins(owed):
    cents = round(owed * 100)
    coins = 0

    while cents > 0:
        if cents >= 25:
            cents -= 25
        elif cents >= 10:
            cents -= 10
        elif cents >= 5:
            cents -= 5
        else:
            cents -= 1
        coins += 1
    return coins


# retuning main function
main()
